### PR TITLE
Add data_format attribute to JaxModel

### PR DESCRIPTION
**Pull request by zimmerrol**

### CODE REVIEW

File: foolbox/models/jax.py [hunk 1]
```diff
@@ -8,7 +8,20 @@
...
+        data_format: Optional[str] = "channels_last",
     ):
         dummy = ep.jax.numpy.zeros(0)
         super().__init__(model, bounds=bounds, dummy=dummy, preprocessing=preprocessing)
+        self._data_format = data_format
+
+    @property
+    def data_format(self) -> str:
+        if self._data_format is None:
+            raise AttributeError(  # AttributeError -> hasattr returns False
```
Comment (jonasrauber) [inline foolbox/models/jax.py:24]: I don't think that's true. Raising an AttributeError inside a function won't change the output of `hasattr` because `hasattr` doesn't execute the function (just tried that).

Comment (zimmerrol) [reply foolbox/models/jax.py:24]: Are you sure @jonasrauber ? The [python documentation](https://docs.python.org/3/library/functions.html#hasattr) for `hasattr` says: _The result is True if the string is the name of one of the object’s attributes, False if not. (This is implemented by calling getattr(object, name) and seeing whether it raises an AttributeError or not.)_

Therefore, I don't understand why `hasattr` shouldn't execute get_attr...

Comment (jonasrauber) [reply foolbox/models/jax.py:24]: True, with the added `@property` it should be fine

**Comment by jonasrauber** [comment]
merged the fixes from #569 … let's see if it helps

**Comment by jonasrauber** [comment]
@zimmerrol What was the motivation for this change?

**Comment by zimmerrol** [comment]
@jonasrauber Some attacks need to know the channel axis and either use the argument passed by the user or infer it automatically from the model. However, this didn't work for JAX models as they had no option to specify the channel axis. Thus, users always had to specify the channel axis for these attacks. 
If I remember correctly I stumbled upon this when I wanted to either increase the coverage of our unit tests for the attacks or implement a new attack with its unit test; when the JAX models don't support the `data_format` attribute one cannot automatically test those attacks on these models.
